src/sentry/tagstore/snuba/backend.py

```diff
@@ -869,6 +871,31 @@ def get_tag_value_paginator_for_projects(
             # If doing a search on semver, we want to hit postgres to query the releases
             return self._get_tag_values_for_semver(projects, environments, query)
 
+        if key == RELEASE_STAGE_ALIAS:
+            organization_id = Project.objects.filter(id=projects[0]).values_list(
+                "organization_id", flat=True
+            )[0]
+            versions = Release.objects.filter_by_stage(
+                organization_id,
+                SearchFilter(
+                    SearchKey(key),
+                    "=",
+                    SearchValue(query),
+                ),
+                project_ids=projects,
+            )
+            if environments:
+                versions = versions.filter(
+                    id__in=ReleaseEnvironment.objects.filter(
+                        environment_id__in=environments
+                    ).values_list("release_id", flat=True)
+                )
+
+            versions = versions.order_by("version").values_list("version", flat=True)[:1000]
+            return SequencePaginator(
+                [(i, TagValue(key, v, None, None, None)) for i, v in enumerate(versions)]
+            )
```

Comment:
Would be good to split this into a separate method like we've done for the other aliases here